titan-server/src/accounts/acl.rs

```diff
@@ -2,35 +2,37 @@ use diesel::{prelude::*, MysqlConnection};
 
 use crate::models;
 use crate::schema;
-
-pub fn get_user_acl(wcf_user_id: i32, wcf_db: &MysqlConnection) -> QueryResult<Vec<models::WcfAclOption>> {
-    let mut acl_option_ids = schema::wcf1_user_to_group::table
-        .inner_join(schema::wcf1_acl_option_to_group::table.on(
-            schema::wcf1_acl_option_to_group::group_id.eq(
-                schema::wcf1_user_to_group::group_id
+use crate::models::WcfUserOptionValue;
+use diesel::mysql::Mysql;
+
+pub fn get_user_acl(wcf_user_id: i32, wcf_db: &MysqlConnection) -> QueryResult<Vec<models::WcfUserGroupOption>> {
+    schema::wcf1_user_group_option::table
+        .select((
+            schema::wcf1_user_group_option::option_id,
+            schema::wcf1_user_group_option::option_name,
+            schema::wcf1_user_group_option::category_name,
+        ))
+        .inner_join(schema::wcf1_user_group_option_value::table.on(
+            schema::wcf1_user_group_option_value::option_id.eq(
+                schema::wcf1_user_group_option::option_id
+            )
+        ))
+        .inner_join(schema::wcf1_user_group::table.on(
+            schema::wcf1_user_group::group_id.eq(
+                schema::wcf1_user_group_option_value::group_id
+            )
+        ))
+        .inner_join(schema::wcf1_user_to_group::table.on(
+            schema::wcf1_user_to_group::group_id.eq(
+                schema::wcf1_user_group::group_id
             )
         ))
-        .select(schema::wcf1_acl_option_to_group::option_id)
         .filter(schema::wcf1_user_to_group::user_id.eq(wcf_user_id))
-        .filter(schema::wcf1_acl_option_to_group::option_value.eq(true))
-        .group_by(schema::wcf1_acl_option_to_group::option_id)
-        .load::<i32>(wcf_db)
-        .expect("failed to load user group acl options");
-
-    let mut user_acl_option_ids = schema::wcf1_acl_option_to_user::table
-        .select(schema::wcf1_acl_option_to_user::option_id)
-        .filter(schema::wcf1_acl_option_to_user::user_id.eq(wcf_user_id))
-        .filter(schema::wcf1_acl_option_to_user::option_value.eq(true))
-        .load::<i32>(wcf_db)
-        .expect("failed to load user acl option ids");
-
-    acl_option_ids.append(&mut user_acl_option_ids);
-    acl_option_ids.sort_unstable();
-    acl_option_ids.dedup();
-
-    schema::wcf1_acl_option::table
-        .filter(schema::wcf1_acl_option::id.eq_any(acl_option_ids))
-        .load::<models::WcfAclOption>(wcf_db)
+        .filter(schema::wcf1_user_group_option_value::option_value.ne("0"))
+        .filter(schema::wcf1_user_group_option_value::option_value.ne(""))
+        .group_by(schema::wcf1_user_group_option::option_id)
+        .order_by(schema::wcf1_user_group_option::option_id)
```

Comment:
> We can further filter this to only objects containing `titan` in the key. Let me know if you think this makes more sense.

Let's go ahead and filter out non-titan ACL options. I don't see titan sharing any forum specific permissions in the near future.

Reply:
Sure, sounds good to me.